state/src/impls/top_level.rs

```diff
@@ -574,7 +574,7 @@ impl TopLevelState {
         Ok((new_root, db, results))
     }
 
-    fn create_shard_level_state(&mut self, owners: Vec<Address>, users: Vec<Address>) -> StateResult<()> {
+    pub fn create_shard_level_state(&mut self, owners: Vec<Address>, users: Vec<Address>) -> StateResult<()> {
```

Comment:
You break the abstraction just for testing.
It should be avoided if possible.

Reply:
Okay, I will put back the unit tests to the original place.